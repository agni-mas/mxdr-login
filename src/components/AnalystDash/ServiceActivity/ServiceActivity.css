.pie-charts {
    margin: 40px 0;
    font-family: sans-serif;
    position: relative;
    display: flex;
    justify-content: space-between;
}

.pie-legend {
    position: absolute;
    top: 30px;
    width: max-content;
    display: flex;
    gap: 20px;
    flex-direction: row;
}

.pie-legend--item {
    margin-bottom: 8px;
    list-style-type: none;
}