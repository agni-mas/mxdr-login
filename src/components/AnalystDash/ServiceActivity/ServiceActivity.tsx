import './ServiceActivity.css'
import react, { FC, useEffect, useState } from 'react';
import Button from '@mui/material/Button';
import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
import PieChartComponent from '../../global/PieChart/PieChart';
import GridItem from '../../layout/GridItem/GridItem';
import { PieChart, Pie, Cell, Label, Tooltip, Legend, Surface, Symbols } from 'recharts';
import { Skeleton } from '@mui/material';
import useFetchSecurityData from '../../../hooks/useFetchSecurityData';


interface ServiceActivity { }

const styles = {
  label: {
    fontFamily: 'Graphik',
    fontStyle: 'normal',
    fontWeight: 300,
    fontSize: 16,
  }
}

const ChartPie = ({ data, label, showLegend }) => {

  const renderLegend = ({ payload }: any) => (
    <ul className='pie-legend'>
      {
        payload.map((entry: any, index: number) => (
          <li className="pie-legend--item" key={index}>
            <Surface width={20} height={10}>
              <Symbols cx={5} cy={5} type="circle" size={100} fill={entry.color} />
            </Surface>
            <span className='legend--item-name'>{entry.value}</span>
          </li>
        ))
      }
    </ul>
  )

  return (
    <PieChart width={145} height={145}>
      <Pie
        dataKey="value"
        data={data}
        innerRadius={50}
        outerRadius={70}
        stroke="black"
      >
        {data.map((d) => (
          <Cell key={d.name} fill={d.fill} />
        ))}
        <Label position="center" fill="white" style={styles.label}>{label}</Label>
      </Pie>
      {/* <Tooltip /> */}
      {showLegend ? (<Legend content={renderLegend} />) : (<></>)}

    </PieChart>
  )
}

const ServiceActivity: FC<ServiceActivity> = () => {
  const [serviceSummaries, setServiceSummaries] = useState([])

  const { data, loading } = useFetchSecurityData('serviceActivity.json')

  useEffect(() => {
    fetchData()
  }, [loading, data])

  const fetchData = () => {
    try {
      let rawData = (data) ?? []
      setServiceSummaries(rawData)
    } catch (err) {
      console.error(err)
    }
  }

  return (
    <GridItem title="Service Activity (Last 7 days)">
      <Button
        disableElevation
        endIcon={<KeyboardArrowDownIcon />}
        color='secondary'
        variant='outlined'
      >
        My Org
      </Button>
      {data ? (<div className='pie-charts'>
        <ChartPie label='Open' data={serviceSummaries} showLegend={true} />
        <ChartPie label='In progress' data={serviceSummaries} showLegend={false} />
        <ChartPie label='Closed' data={serviceSummaries} showLegend={false} />
      </div>) : (<Skeleton variant="rectangular" animation="wave" height={250} />)}

    </GridItem>)
}

export default ServiceActivity;