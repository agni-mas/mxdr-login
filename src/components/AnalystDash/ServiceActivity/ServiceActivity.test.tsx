import { render } from "@testing-library/react";
import ServiceActivity from "./ServiceActivity";

it("renders <ServiceActivity/>", () => {
  const { asFragment } = render(<ServiceActivity />);
  expect(asFragment()).toMatchSnapshot();
});