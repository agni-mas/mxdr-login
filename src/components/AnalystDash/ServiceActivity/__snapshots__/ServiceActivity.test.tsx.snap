// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <ServiceActivity/> 1`] = `
<DocumentFragment>
  <section
    class="item"
  >
    <h2
      class="grid-item-title"
    >
      Service Activity (Last 7 days)
    </h2>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation css-1oovqwt-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      My Org
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="KeyboardArrowDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
      style="height: 250px;"
    />
  </section>
</DocumentFragment>
`;
